### PR TITLE
Add `cnegf` package

### PR DESCRIPTION
**Pull request by steff456**
Part of https://github.com/stdlib-js/todo/issues/1454

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds `cnegf` package
   - [x] JS package
   - [x] C implementation
   - [x] Native addon
   - [x] Tests, examples, readme, etc 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   is part of https://github.com/stdlib-js/todo/issues/1454

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+### Examples
+
+```c
+#include "stdlib/math/base/ops/cnegf.h"
+#include "stdlib/complex/float32.h"
+#include "stdlib/complex/reimf.h"
+#include <stdio.h>
+
+int main() {
+    stdlib_complex64_t x[] = {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
    const stdlib_complex64_t x[] = {
```

We upgraded `cppcheck` (run `make install-deps-cppcheck` locally to upgrade) and it now complains whenever an array which is never mutated is not declared with `const`.

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        stdlib_complex64( 3.14, 1.5 ),
+        stdlib_complex64( -3.14, 1.5 ),
+        stdlib_complex64( 0.0, -0.0 ),
+        stdlib_complex64( 0.0/0.0, 0.0/0.0 )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
        stdlib_complex64( 3.14f, 1.5f ),
        stdlib_complex64( -3.14f, 1.5f ),
        stdlib_complex64( 0.0f, -0.0f ),
        stdlib_complex64( 0.0f/0.0f, 0.0f/0.0f )
```

Single-precision float literals have an `f` suffix.

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    };
+
+    stdlib_complex64_t v;
+    stdlib_complex64_t y;
+    double re;
+    double im;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
    float re;
    float im;
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    int i;
+    for ( i = 0; i < 4; i++ ) {
+        v = x[ i ];
+        stdlib_reimf( v, &re, &im );
+        printf( "z = %lf + %lfi\n", re, im );
+
+        y = stdlib_base_cnegf( v );
+        stdlib_reimf( y, &re, &im );
+        printf( "cnegf(z) = %lf + %lfi\n", re, im );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
        printf( "z = %f + %fi\n", re, im );

        y = stdlib_base_cnegf( v );
        stdlib_reimf( y, &re, &im );
        printf( "cnegf(z) = %f + %fi\n", re, im );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+			b.fail( 'should not be empty' );
+		}
+	}
+	b.toc();
+	if ( !isArray( y ) ) {
+		b.fail( 'should return an array' );
+	}
+	b.pass( 'benchmark finished' );
+	b.end();
+});
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js]:
```suggestion
```

These are no longer needed given the move away from the `( [out], .... )` API.

File: lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.native.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var bench = require( '@stdlib/bench' );
+var uniform = require( '@stdlib/random/base/uniform' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.native.js]:
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.native.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		z = values[ i%values.length ];
+		out = cnegf( z );
+		if ( typeof out !== 'object' ) {
+			b.fail( 'should return an object' );
+		}
+	}
+	b.toc();
+	if ( isnan( realf( out ) ) || isnan( imagf( out ) ) ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.native.js]:
```suggestion
	if ( isnanf( realf( out ) ) || isnanf( imagf( out ) ) ) {
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="@stdlib/types"/>
+
+import { Complex64 } from '@stdlib/types/object';
+
+/**
+* Negates a double-precision complex floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts]:
```suggestion
* Negates a single-precision complex floating-point number.
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+*
+* @param z - complex number
+* @returns result
+*
+* @example
+* var Complex64 = require( `@stdlib/complex/float64` );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts]:
```suggestion
* var Complex64 = require( `@stdlib/complex/float32` );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+* // returns <Complex64>
+*
+* var re = realf( out );
+* // returns 4.2
+*
+* var im = imagf( out );
+* // returns -5.5
+*
+* @example
+* var Complex64 = require( `@stdlib/complex/float64` );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts]:
```suggestion
* var Complex64 = require( `@stdlib/complex/float32` );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+* // returns <Complex64>
+*
+* var re = realf( out );
+* // returns -0.0
+*
+* var im = imagf( out );
+* // returns -0.0
+*
+* @example
+* var Complex64 = require( `@stdlib/complex/float64` );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/index.d.ts]:
```suggestion
* var Complex64 = require( `@stdlib/complex/float32` );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/test.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+* limitations under the License.
+*/
+
+import Complex64 = require( '@stdlib/complex/float32' );
+import cneg = require( './index' );
+
+
+// TESTS //
+
+// The function returns a double-precision complex floating-point number...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/docs/types/test.ts]:
```suggestion
// The function returns a single-precision complex floating-point number...
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/ops/cnegf.h"
+#include "stdlib/complex/float32.h"
+#include "stdlib/complex/reimf.h"
+#include <stdio.h>
+
+int main() {
+	stdlib_complex64_t x[] = {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c]:
```suggestion
	const stdlib_complex64_t x[] = {
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+		stdlib_complex64( 3.14, 1.5 ),
+		stdlib_complex64( -3.14, 1.5 ),
+		stdlib_complex64( 0.0, -0.0 ),
+		stdlib_complex64( 0.0/0.0, 0.0/0.0 )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c]:
```suggestion
		stdlib_complex64( 3.14f, 1.5f ),
		stdlib_complex64( -3.14f, 1.5f ),
		stdlib_complex64( 0.0f, -0.0f ),
		stdlib_complex64( 0.0f/0.0f, 0.0f/0.0f )
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/include/stdlib/math/base/ops/cnegf.h [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Negate a single-precision complex floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/include/stdlib/math/base/ops/cnegf.h:32]:
```suggestion
* Negates a single-precision complex floating-point number.
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+// MODULES //
+
+var Complex64 = require( '@stdlib/complex/float32' );
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Negate a single-precision complex floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* Negates a single-precision complex floating-point number.
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+*
+* @private
+* @param {Complex64} z - complex number
+* @returns {Complex64} result
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float64' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* var Complex64 = require( '@stdlib/complex/float32' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+* // returns { 're': ~4.2, 'im': -5.5 }
+*
+* var re = realf( out );
+* // returns ~4.2
+*
+* var im = imagf( out );
+* // returns -5.5
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float64' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* var Complex64 = require( '@stdlib/complex/float32' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+* @param {Complex64} z - complex number
+* @returns {Complex64} result
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float64' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z1 = new Complex64( -4.2, 5.5 );
+* // returns { 're': -4.2, 'im': 5.5 }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

We should be returning complex number instances here as per L89.

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+* var im = imagf( out );
+* // returns -5.5
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float64' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z2 = new Complex64( 0.0, 0.0 );
+* // returns { 're': 0, 'im': 0 }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+*
+* var re = realf( out );
+* // returns -0.0
+*
+* var im = imagf( out );
+* // returns -0.0
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float64' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* var Complex64 = require( '@stdlib/complex/float32' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z3 = new Complex64( NaN, NaN );
+* // returns { 're': NaN, 'im': NaN }
+*
+* var out = cnegf( z3 );
+* // returns { 're': NaN, 'im': NaN }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/ops/cnegf.h"
+#include "stdlib/complex/float32.h"
+#include "stdlib/complex/reimf.h"
+
+/**
+* Negate a double-precision complex floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* Negates a single-precision complex floating-point number.
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+*
+* @param z        input value
+* @return         result
+*
+* @example
+* #include "stdlib/complex/float32.h"
+* #include "stdlib/complex/real.h"
+* #include "stdlib/complex/imag.h"
+*
+* stdlib_complex64_t z = stdlib_complex64( 3.0, -2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* stdlib_complex64_t z = stdlib_complex64( 3.0f, -2.0f );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+*
+* stdlib_complex64_t out = stdlib_base_cnegf( z );
+*
+* float re = stdlib_real( out );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* float re = stdlib_realf( out );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+/**
+* Negate a double-precision complex floating-point number.
+*
+* @param z        input value
+* @return         result
+*
+* @example
+* #include "stdlib/complex/float32.h"
+* #include "stdlib/complex/real.h"
+* #include "stdlib/complex/imag.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* #include "stdlib/complex/reafl.h"
* #include "stdlib/complex/imagf.h"
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+*
+* stdlib_complex64_t z = stdlib_complex64( 3.0, -2.0 );
+*
+* stdlib_complex64_t out = stdlib_base_cnegf( z );
+*
+* float re = stdlib_real( out );
+* // returns -3.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* // returns -3.0f
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+*
+* float im = stdlib_imag( out );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* float im = stdlib_imagf( out );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+* // returns 2.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* // returns 2.0f
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js [hunk 30]
```diff
@@ -0,0 +1,132 @@
...
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var PINF = require( '@stdlib/constants/float32/pinf' );
+var NINF = require( '@stdlib/constants/float32/ninf' );
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js]:
```suggestion
var isNegativeZerof = require( '@stdlib/math/base/assert/is-negative-zerof' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js [hunk 31]
```diff
@@ -0,0 +1,132 @@
...
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js]:
```suggestion
var isPositiveZerof = require( '@stdlib/math/base/assert/is-positive-zerof' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js [hunk 32]
```diff
@@ -0,0 +1,132 @@
...
+tape( 'the function returns `+0` if provided `-0`', function test( t ) {
+	var actual;
+	var z;
+
+	z = new Complex64( -0.0, -0.0 );
+
+	actual = cnegf( z );
+
+	t.strictEqual( isPositiveZero( realf( actual ) ), true, 'returns expected value' );
+	t.strictEqual( isPositiveZero( imagf( actual ) ), true, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js]:
```suggestion
	t.strictEqual( isPositiveZerof( realf( actual ) ), true, 'returns expected value' );
	t.strictEqual( isPositiveZerof( imagf( actual ) ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js [hunk 33]
```diff
@@ -0,0 +1,132 @@
...
+tape( 'the function returns `-0` if provided `+0`', function test( t ) {
+	var actual;
+	var z;
+
+	z = new Complex64( +0.0, +0.0 );
+
+	actual = cnegf( z );
+
+	t.strictEqual( isNegativeZero( realf( actual ) ), true, 'returns expected value' );
+	t.strictEqual( isNegativeZero( imagf( actual ) ), true, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.js]:
```suggestion
	t.strictEqual( isNegativeZerof( realf( actual ) ), true, 'returns expected value' );
	t.strictEqual( isNegativeZerof( imagf( actual ) ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js [hunk 34]
```diff
@@ -0,0 +1,141 @@
...
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var PINF = require( '@stdlib/constants/float32/pinf' );
+var NINF = require( '@stdlib/constants/float32/ninf' );
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js]:
```suggestion
var isNegativeZerof = require( '@stdlib/math/base/assert/is-negative-zerof' );
var isPositiveZerof = require( '@stdlib/math/base/assert/is-positive-zerof' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js [hunk 35]
```diff
@@ -0,0 +1,141 @@
...
+tape( 'the function returns `+0` if provided `-0`', opts, function test( t ) {
+	var actual;
+	var z;
+
+	z = new Complex64( -0.0, -0.0 );
+
+	actual = cnegf( z );
+
+	t.strictEqual( isPositiveZero( realf( actual ) ), true, 'returns expected value' );
+	t.strictEqual( isPositiveZero( imagf( actual ) ), true, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js]:
```suggestion
	t.strictEqual( isPositiveZerof( realf( actual ) ), true, 'returns expected value' );
	t.strictEqual( isPositiveZerof( imagf( actual ) ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js [hunk 36]
```diff
@@ -0,0 +1,141 @@
...
+tape( 'the function returns `-0` if provided `+0`', opts, function test( t ) {
+	var actual;
+	var z;
+
+	z = new Complex64( +0.0, +0.0 );
+
+	actual = cnegf( z );
+
+	t.strictEqual( isNegativeZero( realf( actual ) ), true, 'returns expected value' );
+	t.strictEqual( isNegativeZero( imagf( actual ) ), true, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/test/test.native.js]:
```suggestion
	t.strictEqual( isNegativeZerof( realf( actual ) ), true, 'returns expected value' );
	t.strictEqual( isNegativeZerof( imagf( actual ) ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 37]
```diff
@@ -0,0 +1,277 @@
...
+#### stdlib_base_cnegf( z )
+
+Negates a single-precision complex floating-point number.
+
+```c
+#include "stdlib/complex/float32.h"
+#include "stdlib/complex/realf.h"
+#include "stdlib/complex/imagf.h"
+
+stdlib_complex64_t z = stdlib_complex64( 3.0, -2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
stdlib_complex64_t z = stdlib_complex64( 3.0f, -2.0f );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 38]
```diff
@@ -0,0 +1,277 @@
...
+
+stdlib_complex64_t out = stdlib_base_cnegf( z );
+
+double re = stdlib_realf( out );
+// returns -3.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
float re = stdlib_realf( out );
// returns -3.0f
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 39]
```diff
@@ -0,0 +1,277 @@
...
+
+double im = stdlib_imagf( out );
+// returns 2.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
float im = stdlib_imagf( out );
// returns 2.0f
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 40]
```diff
@@ -0,0 +1,277 @@
...
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/math/base/special/cabs`][@stdlib/math/base/special/cabs]</span><span class="delimiter">: </span><span class="description">compute an absolute value of a complex number.</span>
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/README.md [hunk 41]
```diff
@@ -0,0 +1,277 @@
...
+<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="links">
+
+<!-- <related-links> -->
+
+[@stdlib/math/base/special/cabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/cabs
+
+<!-- </related-links> -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c [hunk 42]
```diff
@@ -0,0 +1,46 @@
...
+
+	stdlib_complex64_t v;
+	stdlib_complex64_t y;
+	float re;
+	float im;
+	int i;
+	for ( i = 0; i < 4; i++ ) {
+		v = x[ i ];
+		stdlib_reimf( v, &re, &im );
+		printf( "z = %lf + %lfi\n", re, im );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c]:
```suggestion
		printf( "z = %f + %fi\n", re, im );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c [hunk 43]
```diff
@@ -0,0 +1,46 @@
...
+
+		y = stdlib_base_cnegf( v );
+		stdlib_reimf( y, &re, &im );
+		printf( "cnegf(z) = %lf + %lfi\n", re, im );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/examples/c/example.c]:
```suggestion
		printf( "cnegf(z) = %f + %fi\n", re, im );
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 44]
```diff
@@ -0,0 +1,95 @@
...
+* @example
+* var Complex64 = require( '@stdlib/complex/float32' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z1 = new Complex64( -4.2, 5.5 );
+* // returns <Complex64>
+*
+* var out = cnegf( z1 );
+* // returns { 're': ~4.2, 'im': -5.5 }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 45]
```diff
@@ -0,0 +1,95 @@
...
+* @example
+* var Complex64 = require( '@stdlib/complex/float32' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z2 = new Complex64( 0.0, 0.0 );
+* // returns <Complex64>
+*
+* var out = cnegf( z2 );
+* // returns { 're': 0, 'im': 0 }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js [hunk 46]
```diff
@@ -0,0 +1,95 @@
...
+* var im = imagf( out );
+* // returns -0.0
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float32' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var z3 = new Complex64( NaN, NaN );
+* // returns { 're': NaN, 'im': NaN }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/lib/native.js]:
```suggestion
* // returns <Complex64>
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c [hunk 47]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+* Negates a single-precision complex floating-point number.
+*
+* @param z        input value
+* @return         result
+*
+* @example
+* #include "stdlib/complex/float32.h"
+* #include "stdlib/complex/reafl.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/src/main.c]:
```suggestion
* #include "stdlib/complex/realf.h"
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @steff456!

PS. May be good to look through the review comments for future reference.

**Comment by steff456** [comment]
I will! Thanks a lot!

File: lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js [hunk 48]
```diff
@@ -0,0 +1,62 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js [hunk 49]
```diff
@@ -0,0 +1,62 @@
...
+var isArray = require( '@stdlib/assert/is-array' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/ops/cnegf/benchmark/benchmark.js]:
```suggestion
```

**Pull request merged**